files/aws_lifecycle_hooks/attach_ebs_volume.py

```diff
@@ -54,14 +48,34 @@ def attach_volume(
         )
     except botocore.exceptions.ClientError as e:
         if e.response['Error']['Code'] == "VolumeInUse":
-            raise VolumeInUse(e)
+            raise VolumeInUseError(e) from e
         else:
             raise
 
 
-def parse_user_data():
-    # user_data = tools.get_user_data()
-    return 'a', 'b'
+def get_volume_information_from_user_data() -> typing.Tuple[str, str]:
+    """
+    Get the volume id and device name from the user data.
+
+    For this we assume that:
+      - the user_data is parsable (yaml or json)
+      - there is only one one volume
+      - it looks like the following config
+
+    ```yaml
+    ---
+    attach_volumes:
+      - volume_id: volume_id
+        device_name: device_id
+    ```
+
+    :return: A tuple of volume id and device name (in that order)
```

Comment:
For these kind of returns, I usually create a dedicated (data) class to avoid having to remember the order of the tuple.